tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -702,12 +702,27 @@ EOF
 	if [ "${AGENT_INIT}" == "yes" ]; then
 		setup_agent_init "${AGENT_DEST}" "${init}"
 	else
+		info "Setup systemd-base environment for kata-agent"
 		# Setup systemd-based environment for kata-agent
 		mkdir -p "${ROOTFS_DIR}/etc/systemd/system/basic.target.wants"
 		ln -sf "/usr/lib/systemd/system/kata-containers.target" "${ROOTFS_DIR}/etc/systemd/system/basic.target.wants/kata-containers.target"
 		mkdir -p "${ROOTFS_DIR}/etc/systemd/system/kata-containers.target.wants"
 		ln -sf "/usr/lib/systemd/system/dbus.socket" "${ROOTFS_DIR}/etc/systemd/system/kata-containers.target.wants/dbus.socket"
 		chmod g+rx,o+x "${ROOTFS_DIR}"
+
+		if [ "${CONFIDENTIAL_GUEST}" == "yes" ]; then
```

Comment:
Any reason not to do this for non-confidential guests?

Reply:
non-confidential guests do not pull the images inside the podsandbox.  So, no reason to mess around with this on that case.

Reply:
Yeah it still seems a bit fishy to me that the systemd and non-systemd cases have different behavior, but I guess it's safest not to touch the non-confidential case.